tests/MimeTest.php

```diff
@@ -65,4 +73,12 @@ private function streamFromString(string $s)
         rewind($stream);
         return $stream;
     }
+
+    private function invalidStreamFromString(string $s)
+    {
+        $stream = fopen('php://memory', 'w');
+        fwrite($stream, random_bytes(100));
+        fread($stream, 100);
```

Comment:
Hmm, this part is really interesting. Why do you read from the stream after writing to it?

Reply:
Ok, looking further into this, this might be my fault. In #3, I had the following code snippet:

```php
$s = fopen('php://memory', 'w');
fwrite($s, random_bytes(100));
fread($s, 100) -> string(0) ""
```

This does produce an empty string, but not because the stream is write-only! It's because the internal pointer is at `100` after calling `fwrite` with 100 bytes of data. If we `rewind()`, as the above function `streamFromString` does, we find it's perfectly readable:

```php
$s = fopen('php://memory', 'w');
fwrite($s, random_bytes(100));
rewind($s);
fread($s, 1) -> string(1) "�"
```

Changing this line to rewind causes the tests to fail because `UnreadableStreamException` is never thrown. :thinking:

Reply:
You're right. That's interesting.
That's happened because the ```rewind``` function will set the current index of that bytes.
We don't have to concern about this issue because that is the ```rewind``` problem.